### PR TITLE
Akmal / feat: improve input with steppers

### PR DESCRIPTION
**Pull request by akmal-deriv**
- improve input with steppers

### CODE REVIEW

File: lib/components/Input/base/base.scss [hunk 1]
```diff
@@ -118,7 +118,7 @@
             margin-left: var(--core-spacing-200);
         }
         &--disabled {
-            opacity: var(--core-opacity-600);
+            color: var(--component-field-label-color-disabled) !important;
```
Comment (prince-deriv) [inline lib/components/Input/base/base.scss:121]: What is the style we are trying to override and we needed to add `important`. Maybe we can handle the class structure to handle this?

Comment (akmal-deriv) [reply lib/components/Input/base/base.scss:121]: There's default color assigned to text component that I'm trying to overwrite. It always has a priority over any color I pass. I'm thinking it should be fixed by adding color prop to text component or not enforcing any default color that can't be changed without higher priority.

File: lib/components/Input/text-field-with-steppers/index.tsx [hunk 2]
```diff
@@ -7,55 +7,60 @@ import {
...
+        const { value, onChange, disabled, textAlignment, placeholder = 0, unitLeft = '', unitRight = '', decimals = 2, ...rest } = props;
 
         const getFormatValue = (value: number) => parseFloat(value.toFixed(decimals));
 
         const steppersSectionRight = (
             <>
-                {textAlignment !== 'center' && (<button className={clsx("quill-input-steppers-decrement", disabled && 'quill-input-steppers--disabled')}
+                {textAlignment !== 'center' && (<button className={clsx("quill-input-steppers-button", "quill-input-steppers-button--decrement")}
                     onClick={() => {
                         if (!disabled) {
```
Comment (kate-deriv) [inline lib/components/Input/text-field-with-steppers/index.tsx:27]: I'm not sure, bit if button is `disabled` user can't click, so this check might be additional

File: lib/components/Input/base/base.scss (hunk 1 above)
Comment (akmal-deriv) [reply lib/components/Input/base/base.scss:121]: Ok, I noticed there's already color property in the text component, let me try to use that to overwrite it

File: lib/components/Input/text-field-with-steppers/index.tsx [hunk 3]
```diff
@@ -7,55 +7,60 @@ import {
...
+                    >
                     <LabelPairedPlusSmRegularIcon />
                 </button>
             </>
         );
 
         const steppersSectionLeft = (
             <>
-                <button className={clsx("quill-input-steppers-decrement", disabled && 'quill-input-steppers--disabled')}
+                <button className={clsx("quill-input-steppers-button", "quill-input-steppers-button--decrement")}
```
Comment (kate-deriv) [inline lib/components/Input/text-field-with-steppers/index.tsx:54]: Maybe if `jsx` with `button`  is used in multiple places we can create small reusable component?
Like component, which will take as props `className`, `icon` and `- 1` or `+ 1` for `onClick` function calculations?

File: lib/components/Input/text-field-with-steppers/index.tsx (hunk 2 above)
Comment (kate-deriv) [reply lib/components/Input/text-field-with-steppers/index.tsx:27]: Please, ignore if irrelevant

File: lib/components/Input/text-field-with-steppers/text-field-with-steppers.scss [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+            background-color: var(--core-color-opacity-black-100);
+        }
+        &:active:not(:disabled) {
+            background-color: var(--core-color-opacity-black-200);
+        }
+        &:disabled {
+            cursor: not-allowed;
+        }
     }
 }
```
Comment (maryia-deriv) [inline lib/components/Input/text-field-with-steppers/text-field-with-steppers.scss:19]: just a nit-pick:) Do you mind adding a line break at the end of the file to stop Git from complaining about it with ⊖ sign?

File: lib/components/Input/base/base.scss (hunk 1 above)
Comment (akmal-deriv) [reply lib/components/Input/base/base.scss]: I removed this style

File: lib/components/Input/text-field-with-steppers/index.tsx (hunk 3 above)
Comment (akmal-deriv) [reply lib/components/Input/text-field-with-steppers/index.tsx:50]: It's totally not worth it, it's used 2 times in its existence

File: lib/components/Input/text-field-with-steppers/text-field-with-steppers.scss (hunk 4 above)
Comment (akmal-deriv) [reply lib/components/Input/text-field-with-steppers/text-field-with-steppers.scss]: Sure, added

**Review by prince-deriv — APPROVED** [review]

**Review by kate-deriv — APPROVED** [review]

**Review by ahmadtaimoor-deriv — APPROVED** [review]

**Pull request merged**